pyproject.toml

```diff
@@ -209,3 +209,26 @@ exclude= '''
 sdist-only = []
 git-only = [".*", "asv_benchmarks", "azure-pipelines.yml", "benchmarks", "build_tools", "maint_tools"]
 default-ignore = false
+
+[tool.spin]
+package = "sklearn"  # name of your package
+# commands = [
+#   "spin.cmds.meson.test",
+#   "spin.cmds.pip.install",
+# ]
+
+[tool.spin.commands]
+"Build" = [
+  "spin.cmds.pip.install",
+  "spin.cmds.meson.test",
+  "spin.cmds.build.sdist",
+]
+"Documentation" = [
+  "spin.cmds.meson.docs"
+]
+"Environments" = [
+  "spin.cmds.meson.shell",
+  "spin.cmds.meson.ipython",
+  "spin.cmds.meson.python",
+  "spin.cmds.meson.run"
+]
```

Comment:
Do we need these? I'm asking because right now I can install in editable mode and then type `ipython` or `ipython -i some_script.py` or what ever weird flags I'd like to pass to ipython today and it just works. Asked differently: what problem is solved by providing these?

Reply:
If we don't want to use the "out of tree" feature of meson then I agree we can remove it. Is there ever a use-case for doing a "not editable install" while doing development work?